### DISCUSSION TITLE
portable path for macOS

### DISCUSSION THREAD

**Discussion by chriseomi** [wireapp/wire-desktop#5077] [Ideas]
Now that portability is a solid feature for Windows and Linux. Would it be possible to implement portable data path for macOS users too?

Wire.app seems to currently use either the `~/Library/Containers/com.wearezeta.zclient.mac/Data/Library/Application Support/Wire` folder or the `~/Library/Application Support/Wire` folder. Which one  may depend on the type of macOS build used (MAS/dmg), though I'm not sure

**Comment by ekkis** [comment]
there's a `databases/Databases.db` file in there that I can open with SqlLite but seems empty.  where are the conversations kept?

**Comment by chriseomi** [comment]
Not sure. Wire Desktop is an electron app, so maybe Chromium localstorage (wherever that is)?

**Comment by ffflorian** [comment]
@chriseomi did you try starting `/Applications/Wire.app/Contents/macOS/Wire --portable`? With this command, Wire should save your data in the folder you are running it from.

**Comment by chriseomi** [comment]
> @chriseomi did you try starting `/Applications/Wire.app/Contents/macOS/Wire --portable`? With this command, Wire should save your data in the folder you are running it from.

Thanks, I'm testing this out currently and will report back.

**Comment by chriseomi** [comment]
> @chriseomi did you try starting `/Applications/Wire.app/Contents/macOS/Wire --portable`? With this command, Wire should save your data in the folder you are running it from.

So this was quite helpful. However one caveat, is that it appears to always use the folder `Wire.app/Contents/macOS/Data` and not the working directory.

As a result the log file goes nuts saying it doesn't have permissions to do things. Which makes sense cos it's mixing the Data directory into App bundle. 

Also wasn't able to run it as `root`.

Am getting this error

`zsh: illegal hardware instruction  sudo ./Wire --enable-logging --portable`

**Comment by chriseomi** [comment]
I can 100% confirm using `--portable` on macOS means it will use `/Applications/Wire.app/Contents/MacOS/Data`

As evidenced in the log.
```
@wireapp/desktop/main [2021-10-26 23:04:50] Saving user data to "/Applications/Wire.app/Contents/MacOS/Data".
```
 This is despite me launching Wire.app from the commandline whilst in the folder '~/tmp`

**Comment by chriseomi** [comment]
Additionally I tried using the `--user_data_dir="/Volumes/xxxx/tmp/Data"` argument.

This does work.. sort of. However the logging reveals the app is not able to access the folder throwing errors like this

```
[57875:1027/001814.640569:INFO:CONSOLE(69)] "Uncaught (in promise) Error: EPERM: operation not permitted, open '/Volumes/xxxx/tmp/Data/logs/electron.log'"
```

Googling this I though maybe it's a sandbox thing in macOS. And I added Wire.app to Full Disk Access. But that didn't seem to help.

Has anyone else ever managed to successfully run an alternative data folder on macOS?
